jupyterhub/orm.py

```diff
@@ -64,10 +64,12 @@ class Server(Base):
     """
     __tablename__ = 'servers'
     id = Column(Integer, primary_key=True)
+    
+    name = Column(Unicode(32)) # must be unique between user's servers
     proto = Column(Unicode(15), default='http')
     ip = Column(Unicode(255), default='')  # could also be a DNS name
     port = Column(Integer, default=random_port)
-    base_url = Column(Unicode(255), default='/')
+    base_url = Column(Unicode(255), unique=True)
```

Comment:
This column also doesn't need to be unique, and the default should be left here. `Server` is used to store any ip/port combos, so it's only the entirety of ip+port+base_url that would need to be unique, which is not important to enforce at the db level.